plugin.video.espn_3/resources/lib/addon_util.py

```diff
@@ -0,0 +1,282 @@
+import sys
+import urllib
+import base64
+import m3u8
+import time
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+
+import adobe_activate_api
+import util
+import player_config
+import events
+from globals import defaultfanart, pluginhandle, selfAddon, translation, LOG_LEVEL
+from constants import *
+
+TAG = 'Addon_Util: '
+
+def addLink(name, url, iconimage, fanart=None, infoLabels=None):
+    u = sys.argv[0] + '?' + urllib.urlencode(url)
+    liz = xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
+
+    if infoLabels is None:
+        infoLabels={"Title": name}
+    liz.setInfo(type="Video", infoLabels=infoLabels)
+    liz.setProperty('IsPlayable', 'true')
+    liz.setIconImage(iconimage)
+    if fanart is None:
+        fanart=defaultfanart
+    liz.setProperty('fanart_image',fanart)
+    video_streaminfo = dict()
+    liz.addStreamInfo('video', video_streaminfo)
+    ok = xbmcplugin.addDirectoryItem(handle=pluginhandle, url=u, listitem=liz)
+    return ok
+
+
+def addDir(name, url, iconimage, fanart=None, infoLabels=None):
+    u = sys.argv[0] + '?' + urllib.urlencode(url)
+    xbmc.log(TAG + 'Made url to %s' % u, LOG_LEVEL)
+    liz = xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
+    if infoLabels is None:
+        infoLabels={"Title": name}
+    liz.setInfo(type="Video", infoLabels=infoLabels)
+    if fanart is None:
+        fanart=defaultfanart
+    liz.setProperty('fanart_image',fanart)
```

Comment:
In Isengard and above, you should use listItem.setArt() in place of the setProperty for fanart, as well as the icon and thumbnail in the listItem init,. You should also be setting the appropriate 'mediatype' infoLabel for the item.